src/pydata_sphinx_theme/assets/styles/sections/_announcement.scss

```diff
@@ -1,34 +1,38 @@
+.pst-async-banner-revealer {
+  // Setting height to 0 and overflow to hidden allows us to add up the heights
+  // of this element's children before revealing them.
+  height: 0;
+  overflow: hidden;
+
+  // Height to be set by JavaScript, which should trigger the following
+  // transition rule (unless the user has set their system to reduce motion).
+  transition: height 300ms ease-in-out;
+  @media (prefers-reduced-motion) {
+    transition: none;
+  }
+}
+
 .bd-header-version-warning,
 .bd-header-announcement {
+  min-height: 3rem;
   width: 100%;
   display: flex;
   position: relative;
   align-items: center;
   justify-content: center;
   text-align: center;
-  transition: height 300ms ease-in-out;
-  overflow-y: hidden;
   padding: 0.5rem 12.5%; // Horizontal padding so the width is 75%
   // One breakpoint less than $breakpoint-sidebar-primary. See variables/_layout.scss for more info.
   @include media-breakpoint-down(lg) {
     // Announcements can take a bit more width on mobile
     padding: 0.5rem 2%;
   }
 
-  &.init {
-    position: fixed;
-    visibility: hidden;
-  }
-
   p {
     font-weight: bold;
     margin: 0;
   }
 
-  &:empty {
```

Comment:
This functionality—not showing the banner if it's empty—is now taken care of with the changes from #1703 and with the `d-none` class used in this PR.